### PR TITLE
Update Module.spec_to_callback to keep existing specs and copy docs

### PR DESCRIPTION
**Pull request by wojtekmach**
Ref: https://github.com/elixir-lang/ex_doc/issues/942

### CODE REVIEW

File: lib/elixir/lib/kernel/typespec.ex [hunk 1]
```diff
@@ -75,12 +75,21 @@ defmodule Kernel.Typespec do
...
+    {set, bag} = :elixir_module.data_tables(module)
 
     filter = fn {:spec, expr, pos} ->
       if spec_to_signature(expr) == signature do
-        delete_typespec(bag, :spec, expr, pos)
-        store_typespec(bag, :callback, expr, pos)
+        kind = :callback
+        store_typespec(bag, kind, expr, pos)
+
+        case :ets.lookup(set, {:function, name, arity}) do
```
Comment (fertapric) [inline lib/elixir/lib/kernel/typespec.ex]: I tried locally just by removing the line `delete_typespec(bag, :spec, expr, pos)` and the `@spec` is shown on IEx, why is this part needed? 🙂

**Comment by josevalim** [comment]
That part is needed to add the docs to the callback. Btw Wojtek, we need a
test that the spec was not deleted. It could be one extra assertion in the
test you added. Although we should probably add unit tests to
typespec_test.exs, at least where specs/callbacks are concerned.
-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Director of R&D*

**Comment by fertapric** [comment]
@josevalim Ah! Gotcha. Thanks :)

Btw, I would update update this comment: 

https://github.com/elixir-lang/elixir/blob/985bfe53b4074db34d548e6856ac432e699726c0/lib/elixir/lib/protocol.ex#L45-L48

File: lib/elixir/lib/module.ex [hunk 2]
```diff
@@ -974,9 +974,9 @@ defmodule Module do
   end
 
   @doc """
-  Converts the given spec to a callback.
+  Copies the given spec as a callback.
```
Comment (fertapric) [inline lib/elixir/lib/module.ex]: Maybe mention that the docs are also copied to the callback?

File: lib/elixir/lib/module.ex [hunk 3]
```diff
@@ -974,9 +974,11 @@ defmodule Module do
...
 
-  Returns `true` if there is such a spec and it was converted to a callback.
+  Returns `true` if there is such a spec and it was copied as a callback.
+  The function associated to the spec has documentation defined prior to
```
Comment (fertapric) [inline lib/elixir/lib/module.ex]:
```suggestion
  If the function associated to the spec has documentation defined prior to
```

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:
